server/src/main/java/org/apache/cassandra/sidecar/cdc/CdcConfigImpl.java

```diff
@@ -0,0 +1,362 @@
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Implementation of the interface {@link CdcConfig}, an in-memory representation holding
+ * CDC and Kafka configurations from "configs" table inside sidecar internal keyspace.
+ */
+@Singleton
+public class CdcConfigImpl implements CdcConfig
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(CdcConfigImpl.class);
+    private static final int DEFAULT_MAX_WATERMARKER_SIZE = 400000;
+    private static final String DEFAULT_JOB_ID = "test-job-id";
+    private static final int DEFAULT_MAX_COMMITLOGS_PER_INSTANCE = 4;
+    private static final int DEFAULT_MAX_RECORD_BYTE_SIZE = -1;
+    private final SchemaKeyspaceConfiguration schemaKeyspaceConfiguration;
+    private final CdcConfiguration cdcConfiguration;
+    private final CdcConfigAccessor cdcConfigAccessor;
+    private final KafkaConfigAccessor kafkaConfigAccessor;
+    private final List<ThrowingRunnable> configChangeListeners = Collections.synchronizedList(new ArrayList<>());
+    private final ConfigRefreshNotifier configRefreshNotifier;
+    private volatile Map<String, String> kafkaConfigMappings = Map.of();
+    private volatile Map<String, String> cdcConfigMappings = Map.of();
+
+    @Inject
+    public CdcConfigImpl(CdcConfiguration cdcConfiguration,
+                         SchemaKeyspaceConfiguration schemaKeyspaceConfiguration,
+                         CdcConfigAccessor cdcConfigAccessor,
+                         KafkaConfigAccessor kafkaConfigAccessor,
+                         PeriodicTaskExecutor periodicTaskExecutor)
+    {
+        this.schemaKeyspaceConfiguration = schemaKeyspaceConfiguration;
+        this.cdcConfiguration = cdcConfiguration;
+        this.cdcConfigAccessor = cdcConfigAccessor;
+        this.kafkaConfigAccessor = kafkaConfigAccessor;
+
+        if (this.schemaKeyspaceConfiguration.isEnabled())
+        {
+            this.configRefreshNotifier = new ConfigRefreshNotifier();
+            periodicTaskExecutor.schedule(configRefreshNotifier);
+        }
+        else
+        {
+            this.configRefreshNotifier = null;
+        }
+    }
+
+    @Override
+    public Map<String, Object> kafkaConfigs()
+    {
+        Map<String, Object> kafkaConfigs = new HashMap<>();
+        kafkaConfigs.putAll(getAuthConfigs());
+        kafkaConfigs.putAll(kafkaConfigMappings);
+        return ImmutableMap.copyOf(kafkaConfigs);
+    }
+
+    @Override
+    public Map<String, Object> cdcConfigs()
+    {
+        return ImmutableMap.copyOf(cdcConfigMappings);
+    }
+
+    @Override
+    public boolean isConfigReady()
+    {
+        return cdcConfigAccessor.isSchemaInitialized()
+                && !kafkaConfigMappings.isEmpty()
+                && !cdcConfigMappings.isEmpty();
+    }
+
+    @Override
+    public String kafkaTopic()
+    {
+        return cdcConfigMappings.getOrDefault(ConfigKeys.TOPIC.lowcaseName, null);
```

Comment:
NIT, there's no need to default to null here
```suggestion
        return cdcConfigMappings.get(ConfigKeys.TOPIC.lowcaseName);
```